keps/sig-node/1995-container-notifier/README.md

```diff
@@ -0,0 +1,899 @@
+
+    // State represents the current stage of the PodNotification.
+    // Default to PodNotificationStateNew.
+    State PodNotificationStateType
+
+    // Error represents the failing reason of the PodNotification.
+    // +optional
+    Error *PodNotificationError
+}
+
+// PodNotificationStateType represents the current stage of the PodNotification.
+type PodNotificationStateType string
+const(
+  // PodNotificationStateNew suggests the PodNotification is newly created and
+  // has not been picked up yet.
+  PodNotificationStateNew PodNotificationStateType = "New"
+
+  // PodNotificationStateSucceeded suggests the PodNotidication has been
+  // successfully executed against all qualified Containers.
+  PodNotificationStateSucceeded PodNotificationStateType = "Succeeded"
+
+  // PodNotificationStateFailed suggests the PodNotidication has been executed
+  // however at least one of the qualified Containers returned an error. Details
+  // can be found in 'Containers'.
+  PodNotificationStateFailed PodNotificationStateType = "Failed"
+)
+
+// ContainerNotificationStatus represents the execution state of the handler in
+// the requested ContainerNotifier.
+type ContainerNotificationStatus struct {
+    // The name of the Container in which the handler has been executed.
+    // Required.
+    Name string
+
+    // If set, it suggests the timestamp at when the first time the execution
+    // of the action has started in UTC.
+    // +optional
+    StartTime *metav1.Time
+
+    // If set, it suggests the timestamp at when the first time the execution
+    // of the handler has completed in UTC.
+    // Note that `CompleteTime` does NOT mean the handler has been executed
+    // successfully. Refer to `Succeeded` field for more details.
+    // +optional
+    CompleteTime *metav1.Time
+
+    // Succeeded will be set to true when the handler has been executed
+    // successfully, and it will be set to false otherwise. It will not be
+    // specified if the status of execution is unknown(i.e., in execution).
+    // +optional
+    Succeeded *bool
+
+    // If specified, Error represents the last error encountered when executing
+    // the handler.
+    // +optional
+    Error *PodNotificationError
+}
+
+type PodNotificationError struct {
+    // Type of the error
+    // Required.
+    Type PodNotificationErrorType
+
+    // A human-readable message indicating details why the action failed.
+    Message string
+
+    // It indicates when the error occurred
+    Timestamp metav1.Time
+}
+
+// PodNotificationErrorType represents the type of error occurred in a
+// Notifiication.
+type PodNotificationErrorType string
+
+// More error types could be added, e.g., Forbidden, Unauthorized, etc.
+const (
+    // Pod not found error
+    PodNotificationErrorPodNotFound PodNotificationErrorType = "PodNotFound"
+
+    // The execution of a handler timed out.
+    PodNotificationErrorHandlerTimeout PodNotificationErrorType = "HandlerTimeout"
```

Comment:
yes, it will be surfaced both in PodNotificationStatus aggregated field "State" as well as detailed in "Containers" which users can tell in which container the notifier failed.